### DISCUSSION TITLE
Midnight commander started from tsu shows distorted/unaligned symbols (not sure it's a bug)

### DISCUSSION THREAD

**Discussion by Codereamp** [termux/termux-packages#18001] [Q&A]
Windows/putty
**mc -a** 

run after the connection

![normal](https://github.com/termux/termux-packages/assets/41295076/ca366729-2ee1-435a-8dba-d026bde429a6)

after tsu

![tsu-distorted](https://github.com/termux/termux-packages/assets/41295076/8afc28aa-cc19-4fbf-9803-f3bed9432a23)

I'm not sure this is an issue to be reported, probably something to do with encoding/font/shell settings.

**Comment by sylirre** [comment]
Try running `reset` or `stty sane` after `su`. Pretty much looks like `su` bug. Had a similar terminal mess up issue with addonsu from Lineage OS.

Termux doesn't change console settings on its own. They are controlled by a running utility. Not all utilities set up them correctly when creating a new `/dev/pts/X` instance.

**Reply by Codereamp** [reply]
sylirre, thanks for the suggestion, unfortunately it didn't help, both commands changed nothing.
additional information.

- My post was about accessing with putty, now I've did this on the device itself (nook with eInk screen). Actually in termux the drawing after tsu was ok though barely visible with the current grayscale palette. 
- The effects of invalid drawing are different if one looks. Sometimes it's black blocks and unaligned drawing like in the screenshots. But also it's non-drawing at invalidated areas like when moving with menu I see the pieces of previous menus still visible. 
![menu](https://github.com/termux/termux-packages/assets/41295076/35aba768-8872-4363-b3cc-7ce396f7ba06)

- searching by looking at googled images I found some screenshots similar to mine, like in [this old ticket](https://midnight-commander.org/ticket/3048). The discussion was about different TERM values like xterm and xterm-256color. I tried to play with them accessing from the comp, but all the same, the terminal always looked colored-ehnanced, so porbably xterm-256color regardless of the settings in the putty and in the terminal. (As I understand in tsu it's even hard-coded to pass TERM=xterm-256color)

**Reply by sylirre** [reply]
xterm is wrong terminal value for putty. You can use only `putty-256color` or `putty`.

Looks like `tsu` hardcodes the value. Report issue to https://github.com/cswl/tsu as we didn't developed `tsu` on our own.

Normally it should pick existing `TERM` variable and only use `xterm-256color` as fallback.

**Reply by Codereamp** [reply]
Thanks for your help. Long story short, the problem solved, it was the putty version 0.70 that for some reasons incorrectly rendered all variants of configurations. I found that the latest is 0.79 and this one gave the correct rendering. 

Just for reference, tsu has --dbg option and outputs exactly what it feed to su. So I started with the sh file containing the command line and tried to modify, even b&w chain led to the bad result (xterm).
